### DISCUSSION TITLE
How to post a http request to server side based DGS?

### DISCUSSION THREAD

**Discussion by nandychen** [Netflix/dgs-framework#825] [General]
For example: there is a client named A which is developed with javascript，and a server named B developed with DGS。Now i want to send a post request to query some data from B,  the question is how should we handle the incoming http request? I found there no any code samples like controllers in spring mvc, the methods annotated with @DgsData or @DgsQuery cannot interact with the springmvc request.

**Comment by srinivasankavitha** [comment]
Assuming you are posting a graphql request to server B, the DGS framework
will handle that under the hood. All you need to do is implement the data
fetcher, i.e. a method annotated with @DgsQuery or @DgsMutation as
applicable for the request. The framework is built on top of graphql-java
and the request is handled field by field by invoking  the appropriate data
fetchers that you have implemented for each field:
https://www.graphql-java.com/tutorials/getting-started-with-spring-boot
The DGS framework provides ***@***.***` annotation as a shortcut way to wire
up the data fetchers: https://netflix.github.io/dgs/getting-started/

Hope this helps.

**Reply by nandychen** [reply]
Hi SrinivasanKavitha, thank you so much for the reply. It's very helpful to quick start on DGS. One more thing, i want to ask for advice for this problem: 

The client A stated above is an app developed with vue, and the server B is an api-gateway which accept the http request posted from A, and then dispatch the request to downstream server C(provide restful api) which act as the DataFetcher, for example:

step 1 - [Post request from A]: 

curl -X POST http://127.0.0.1:8080/graphql \
  -H 'Content-Type: application/json' \
  -H 'x-uid: 1' \
  -H 'x-openid: xxx' \
  -d '{"query": "query GetGoodsDetails($spuId: ID!) {\n  getGoodsDetails(spuId: $spuId) {\n    spu {\n      spuId\n      name\n      stock\n    }\n    merchant {\n      merchantId\n      name\n    }\n    brandInfo {\n      id\n      name\n      logo\n    }\n  }\n}","variables":{"spuId":"xxx123"},"operationName":"GetGoodsDetails"}'

step 2 - [Accept the request above and parse the `query`,`operationName`,`variables` in server B, and then next step]

step 3 - [Dispatch the request to downstream server C to get data accorded with the dynamic GraphQL schemas parsed in server B] 

My question is: Is this pattern appropriate to integrate with downstream servers? is there any performance issues? any solutions in DGS?

Thank you!
